sound/hda/intel-dsp-config.c

```diff
@@ -321,6 +321,18 @@ static const struct config_entry config_table[] = {
 	},
 #endif
 
+/* Alder Lake */
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_ALDERLAKE)
+	{
+		.flags = FLAG_SOF | FLAG_SOF_ONLY_IF_DMIC_OR_SOUNDWIRE,
+		.device = 0x7ad0,
+	},
+	{
+		.flags = FLAG_SOF | FLAG_SOF_ONLY_IF_DMIC_OR_SOUNDWIRE,
+		.device = 0x51c8,
+	},
```

Comment:
I think there were patches recently, enabling SOF on TGL unconditionally since SST doesn't support TGL. Does SST support ADL?

Reply:
@lyakh We talked about is TGL+, so I think NO for ADL here.

Reply:
> I think there were patches recently, enabling SOF on TGL unconditionally since SST doesn't support TGL. Does SST support ADL?

My bad, #2665 is confusing people. @lyakh Legacy HDA driver should still be the default even if SST doesn't support ADL. See https://github.com/thesofproject/linux/pull/2672

Reply:
Ack @bardliao @xiulipan , this is still needed to allow to select between non-DSP and DSP drivers.

It would be nice to avoid repeating all the PCI IDs here, but unless we change approach, this is not needed.